### DISCUSSION TITLE
dag import error when using sys.path.append(str(Path('.').absolute().parent))

### DISCUSSION THREAD

**Discussion by ezerkar** [apache/airflow#17718] [Q&A]
**Apache Airflow version**:

v2.1.2

**OS**:

Ubuntu 20.04.1 LTS 

**Deployment**:

local

**What happened**:

have a file in a parent directory I want to import from, so did this:
```
import sys
from pathlib import Path
sys.path.append(str(Path('.').absolute().parent))
from get_data_functions import *
```
this seem to be working in general python but airflow gives out a DAG import error:
`ModuleNotFoundError: No module named 'get_data_functions'`

**Are you willing to submit a PR?**
No idea how to fix this, but if you give me some hints I could try

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
This is not airflow issue - I converted it to a discussion. Please, in the future, if you have similar questions open them as discussions  - or best, in case of general python questions, try to debug on it's own a bit and ask Google. The Issues are for Airlfow issues nor for general python questions, it's quite a bit spaming everyone if you try to use it this way.

This is general python question. Make sure you have `__init__.py` in the folder you import and double check (by printing) whether your folder is correct. Just do some debugging. It will work eventually.

**Comment by ezerkar** [comment]
a. sorry, and apologies for any problems caused
b. hmm, AFAIK the python files work when run outside airflow, but airflow
gives an error message. think would be nice to make it work

**Reply by potiuk** [reply]
Have you read? https://airflow.apache.org/docs/apache-airflow/stable/modules_management.html  - it describes how to import in Airflow in details. You should follow this

**Comment by ezerkar** [comment]
Thanks!!

Was not aware. So any use of a package added by sys.path.append in the code
itself will raise an error in airflow as airflow preinspects the python
path before running the code? and that's intended?

בתאריך יום ה׳, 19 באוג׳ 2021, 22:33, מאת Jarek Potiuk ‏<
***@***.***>:

**Reply by yijianxilai** [reply]
the same problem, how do you fix it finally？thanks a lot
